akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -89,6 +90,12 @@ private[remote] class Encoder(
 
       override protected def logSource = classOf[Encoder]
 
+      private var debugLogEnabled = false
+
+      override def preStart(): Unit = {
+        debugLogEnabled = log.isDebugEnabled
```

Comment:
couldn't this do the check of the other boolean as well, as they wont change after this?